azure/network-security-group/main.tf

```diff
@@ -112,4 +112,6 @@ resource "azurerm_network_watcher_flow_log" "main" {
       days    = var.flow_log_config["retention_days"]
     }
   }
+
+  depends_on = [azurerm_network_security_group.main]
```

Comment:
Its on the destroy that the problem arrises.  Creating the resource isn't a problem.

Reply:
Something feels amiss here, I've asked @tb-ans for a second opinion

Reply:
Here's the error for reference

│ Error: retrieving Flow Log (Subscription: "a1b49583-2f58-47a2-b3c3-b390970e7b7c"
│ Resource Group Name: "rg-tfmex-adv-spoke-nw-uksouth"
│ Network Watcher Name: "nw-tfmex-adv-spoke-uksouth"
│ Flow Log Name: "fl-tfmex-adv-spoke"): unexpected status 404 with error: ResourceNotFound: The Resource 'Microsoft.Network/networkWatchers/nw-tfmex-adv-spoke-uksouth/flowLogs/fl-tfmex-adv-spoke' under resource group 'rg-tfmex-adv-spoke-nw-uksouth' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix

Reply:
Not really sure how the error links to an explicit dependency since it's complaining about the flowlog not existing, unless I'm misreading something.

Have you tried a full apply/destroy with fresh state?

Reply:
@tb-ans Yes, this is after a clean deploy and destroy 15 mins after
The error doesn't present when the depends_on is added

Reply:
Created issue for further investigation: https://github.com/ans-coe/terraform-modules/issues/102